main.go

```diff
@@ -168,11 +168,15 @@ func Copy(client *ssh.Client) {
 }
 
 // ConfigureAuthentication configures the authentication method.
-func ConfigureAuthentication(key string, password string) []ssh.AuthMethod {
+func ConfigureAuthentication(key string, passphrase string, password string) []ssh.AuthMethod {
 	// Create signer for public key authentication method.
 	auth := make([]ssh.AuthMethod, 1)
 	if key != "" {
-		targetSigner, err := ssh.ParsePrivateKey([]byte(key))
+		if passphrase != "" {
```

Comment:
@Briaker If you put the following above this if statement you should be fine:

```go
var err error
var targetSigner ssh.AuthMethod
```

Afterwards, you might then just need to replace the `:=` with `=`, because the variables are declared in the upper scope by the newly added `var` statement.

Reply:
Ahhh, I a rookie mistake haha. You're right `err` and `targetSigner` needed to be initialized outside of the if for the build to pass. I also used `ssh.Signer` instead of `ssh.AuthMethod` for `targetSigner`. I'm testing it now, my build was successful, just want to run it once in a workflow to make sure that the passphrase is actually working. Ill push my changes shortly.

Reply:
Mistakes are the key to any learning process! Embrace them! 😉

Reply:
Ok updated the code, and built and tested it in a custom workflow. Took a bit, mostly to figure out what kind of fignerprint I needed to use. 
But it should be all good now, I was able to connect to a remote host using a passphrase ssh key.

Also did a quick rebase to undo a debug change I made.